src/components/common/YourCompany/ChangeBusinessType.vue

```diff
@@ -397,6 +392,64 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
   protected submitTypeChange () {
     this.setEntityType(this.selectedEntityType)
     this.isEditingType = false
+
+    if (this.shouldUpdateName()) {
+      const originalName = this.getEntitySnapshot?.businessInfo.legalName
+      const updatedName = this.getUpdatedName(originalName)
+
+      if (originalName !== updatedName) {
+        const nameRequest = {
+          legalType: this.selectedEntityType,
+          legalName: updatedName,
+          nrNumber: this.getEntitySnapshot?.businessInfo?.nrNumber
+        }
+
+        this.setNameRequest(nameRequest)
+        this.setNameChangedByType(true)
+      }
+    }
+  }
+
+  private shouldUpdateName (): boolean {
+    return this.isNumberedCompany && !this.hasBusinessNameChanged
+  }
+
+  private getUpdatedName (originalName: string): string {
+    if (this.isUnlimitedLiability || this.isCommunityContribution) {
+      return originalName.endsWith(' LTD.') ? originalName : originalName + ' LTD.'
```

Comment:
Are there more patterns to catch than just LTD?

Reply:
just LTD, I confirmed with @ethantspitt